field/src/types.rs

```diff
@@ -490,6 +490,19 @@ pub trait Field64: Field {
     // TODO: Move to `Field`.
     fn from_noncanonical_u64(n: u64) -> Self;
 
+    /// Returns `n` as an element of this field. Assumes that `|n| < Self::ORDER`.
+    // TODO: Should probably be unsafe.
+    // TODO: Move to `Field`.
+    #[inline]
+    fn from_canonical_i64(n: i64) -> Self {
+        Self::from_canonical_u64(if n < 0 {
+            // Assumption guarantees that -n < ORDER.
+            Self::ORDER - (-n as u64)
```

Comment:
Subtle bug: this code fails when `n` = -2^63. In that case, `-n` overflows since 2^63 is not in the range of `i64`.

Reply:
One way to do this is:
```suggestion
            Self::ORDER.wrapping_add(n as u64)
```

Reply:
Good catch!